src/Uno.UI/UI/Xaml/Controls/Layouter/Layouter.Android.cs

```diff
@@ -28,6 +28,13 @@ protected Size MeasureChildOverride(View view, Size slotSize)
 			var heightSpec = ViewHelper.SpecFromLogicalSize(slotSize.Height);
 
 			var needsForceLayout = double.IsPositiveInfinity(slotSize.Width) || double.IsPositiveInfinity(slotSize.Height);
+			if (!needsForceLayout && FeatureConfiguration.FrameworkElement.InvalidateNativeCacheOnRemeasure)
+			{
+				// uno12315: ensure the native measure cache is not used when measure-spec has changed since.
+				needsForceLayout |=
+					view.MeasuredWidth != ViewHelper.LogicalToPhysicalPixels(slotSize.Width) ||
+					view.MeasuredHeight != ViewHelper.LogicalToPhysicalPixels(slotSize.Height);
```

Comment:
Do we want to make this `!DoubleUtil.AreClose(..., ...)`?

Reply:
this is actually not needed, since we are actually comparing `int` on both sides